test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -12,234 +12,235 @@
 				}
 			}
 		}
 	}
 	return nil
 }
 
-// The Wrapper function createPodForSecondaryNetwork creates the Pod adding the annotation, arguments, commands, Node, container name,
-// resource requests and limits as arguments with the NewPodBuilder API
-func (data *TestData) createPodForSecondaryNetwork(ns string, nodeName string, podNum int, testPodName string, resNum int64) error {
-	computeResources := resource.NewQuantity(resNum, resource.DecimalSI)
-	return antreae2e.NewPodBuilder(podData[podNum].nameOfPods, ns, busyboxImage).OnNode(nodeName).WithContainerName(ctrName).WithCommand([]string{"sleep", "infinity"}).WithAnnotations(
-		map[string]string{
-			"k8s.v1.cni.cncf.io/networks": fmt.Sprintf("%s", data.formAnnotationStringOfPod(podNum)),
-		}).WithLabels(
-		map[string]string{
-			"App": fmt.Sprintf("%s", testPodName),
-		}).WithResources(corev1.ResourceList{reqName: *computeResources}, corev1.ResourceList{reqName: *computeResources}).Create(data.e2eTestData)
-}
-
-func TestNativeSecondaryNetwork(t *testing.T) {
-	// once the setupTestWithSecondaryNetworkConfig is successful, we have all the prerequisites enabled and running.
-	_, err := testData.setupTestWithSecondaryNetworkConfig(t)
+func testSecondaryNetwork(t *testing.T, networkType string, pods []*testPodInfo) {
+	e2eTestData, err := antreae2e.SetupTest(t)
 	if err != nil {
-		t.Logf("Error when setupTestWithSecondaryNetworkConfig: %v", err)
+		t.Fatalf("Error when setting up test: %v", err)
 	}
+	defer antreae2e.TeardownTest(t, e2eTestData)
+
+	testData := &testData{e2eTestData: e2eTestData, networkType: networkType, pods: pods}
+
 	t.Run("testCreateTestPodOnNode", func(t *testing.T) {
-		testData.createPodOnNode(t, nameSpace, clusterInfo.controlPlaneNodeName)
+		testData.createPods(t, e2eTestData.GetTestNamespace())
 	})
 	t.Run("testpingBetweenInterfaces", func(t *testing.T) {
 		err := testData.pingBetweenInterfaces(t)
 		if err != nil {
-			t.Logf("Error when pinging between interfaces: %v", err)
+			t.Fatalf("Error when pinging between interfaces: %v", err)
 		}
 	})
 }
+
+func TestSriovNetwork(t *testing.T) {
+	// Create Pods on the control plane Node, assuming a single Node cluster for the SR-IOV
+	// test.
+	controlPlaneNodeName := antreae2e.NodeName(0)
+	pods := []*testPodInfo{
+		{
+			podName:           "sriov-pod1",
+			nodeName:          controlPlaneNodeName,
+			interfaceNetworks: map[string]string{"eth1": "sriov-net1", "eth2": "sriov-net2"},
+		},
+		{
+			podName:           "sriov-pod2",
+			nodeName:          controlPlaneNodeName,
+			interfaceNetworks: map[string]string{"eth2": "sriov-net1", "eth3": "sriov-net3"},
+		},
+		{
+			podName:           "sriov-pod3",
+			nodeName:          controlPlaneNodeName,
+			interfaceNetworks: map[string]string{"eth4": "sriov-net1"},
+		},
+	}
+	testSecondaryNetwork(t, networkTypeSriov, pods)
+}
+
+func TestVLANNetwork(t *testing.T) {
+	node1 := antreae2e.NodeName(1)
+	node2 := antreae2e.NodeName(2)
```

Comment:
maybe you want to fail the test if not enough Nodes?

Reply:
Yes. Would you suggest to decide which Nodes to use based on the cluster size? E.g. use Node 1 for all Pods if the cluster has only two Nodes (with the 1st Node to be the control plane Node), and use Node 0 for a single Node cluster.

Reply:
I would personally just use `NodeName(0)` and `NodeName(1)` here, and fail the test immediately with `t.Fail` if `clusterInfo.numNodes < 2`.

Reply:
Done.